dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/cluster/ClusterManager.java

```diff
@@ -48,12 +49,15 @@ public class ClusterManager {
     @Autowired
     private RegistryClient registryClient;
 
+    private WorkerGroupTaskDispatchManager workerGroupTaskDispatchManager;
+
     public ClusterManager() {
         this.masterClusters = new MasterClusters();
         this.workerClusters = new WorkerClusters();
     }
 
-    public void start() {
+    public void start(WorkerGroupTaskDispatchManager workerGroupTaskDispatchManager) {
+        this.workerGroupTaskDispatchManager = workerGroupTaskDispatchManager;
```

Comment:
You should add a method to register the `WorkerGroupListener` rather than change the start method, this is too strange.

Reply:
i will add registerWorkerGroupListener function